packages/core/content-manager/server/src/services/document-manager.ts

```diff
@@ -196,85 +196,30 @@ const documentManager = ({ strapi }: { strapi: Core.Strapi }) => {
         .then((result) => result?.versions.at(0));
     },
 
-    async publishMany(entities: Document[], uid: UID.ContentType) {
-      if (!entities.length) {
-        return null;
-      }
-
-      // Validate entities before publishing, throw if invalid
-      await Promise.all(
-        entities.map((document: Document) => {
-          return strapi.entityValidator.validateEntityCreation(
-            strapi.getModel(uid),
-            document,
-            undefined,
-            // @ts-expect-error - FIXME: entity here is unnecessary
-            document
-          );
-        })
-      );
-
-      // Only publish entities without a published_at date
-      const entitiesToPublish = entities
-        .filter((doc: Document) => !doc[PUBLISHED_AT_ATTRIBUTE])
-        .map((doc: Document) => doc.id);
-
-      const filters = { id: { $in: entitiesToPublish } };
-      const data = { [PUBLISHED_AT_ATTRIBUTE]: new Date() };
-      const populate = await buildDeepPopulate(uid);
-
-      // Everything is valid, publish
-      const publishedEntitiesCount = await strapi.db.query(uid).updateMany({
-        where: filters,
-        data,
-      });
-      // Get the updated entities since updateMany only returns the count
-      const publishedEntities = await strapi.db.query(uid).findMany({
-        where: filters,
-        populate,
+    async publishMany(
```

Comment:
I think we'd be losing the event emission with this change?

Also FYI (I think I can deal with the conflict) with bulk locale actions this is
something like:

`async publishMany(uid: UID.ContentType, documentIds: string[], locale?: string| string[])`

e.g. it can take multiple locales. You can see it here

https://github.com/strapi/strapi/blob/65c408613ecd61094a501a8537ed6c87e1498793/packages/core/content-manager/server/src/services/document-manager.ts#L200

Reply:
webhooks events are going to be triggered at document service level, Marc is currently working on that [here](https://github.com/strapi/strapi/pull/20129)

Reply:
Thanks, I think I can remove that from my PR too then